relay-metrics/src/aggregation.rs

```diff
@@ -1388,8 +1420,8 @@ impl Aggregator {
             self.buckets.retain(|key, entry| {
                 if force || entry.elapsed() {
                     // Take the value and leave a placeholder behind. It'll be removed right after.
-                    let value = std::mem::replace(&mut entry.value, BucketValue::Counter(0.0));
-                    cost_tracker.subtract_cost(key.project_key, value.cost());
+                    let value = mem::replace(&mut entry.value, BucketValue::Counter(0.0));
+                    cost_tracker.subtract_cost(key.project_key, key.cost() + value.cost());
```

Comment:
can we call subtract_cost twice instead? then overflow/underflow handling is also dealt with in one place